src/Zicht/Itertools/reductions.php

```diff
@@ -124,20 +124,16 @@ function join($glue = '')
  * Returns a closure that chains lists together
  *
  * > $lists = [[1, 2, 3], [4, 5, 6]]
- * > iterable($lists)->reduce(reductions\chain())
+ * > iterable($lists)->reduce(reductions\chain(), new ChainIterator())
  * results in a ChainIterator: 1, 2, 3, 4, 5, 6
  *
  * @return \Closure
  */
 function chain()
 {
-    $chainIterator = new ChainIterator();
```

Comment:
This is the problem I am trying to solve... 

This `new ChainIterator` should not be here, but instead it makes much more sense if this is passed through the `$initializer` parameter of `reduce`.  That way the first parameter of the reduction closure is always the ChainIterator.